classes/schedules/ActionScheduler_NullSchedule.php

```diff
@@ -5,6 +5,9 @@
  */
 class ActionScheduler_NullSchedule extends ActionScheduler_SimpleSchedule {
 
+	/** @var DateTime|null */
+	public $scheduled_date;
```

Comment:
There are quite a few layers of inheritance here, but I have a suspicion that the constructor is assuming it can write to the property of the same name that exists in superclass [`ActionScheduler_Abstract_Schedule`](https://github.com/woocommerce/action-scheduler/blob/3.5.3/classes/abstracts/ActionScheduler_Abstract_Schedule.php#L13), which is private (and perhaps this was possible in much earlier versions of PHP?).

With that in mind, I think it may make more sense to resolve this by making `ActionScheduler_Abstract_Schedule::scheduled_date` protected, vs exposing a new public property. What do you think?

Reply:
...Also, our apologies for the delay in getting to this PR: if it's helpful/if you're focused on other things, we can drive this forward.

Reply:
Thanks for the review, @barryhughes. I wasn't sure what would be the best in terms of the visibility of the property. I went ahead an pushed a new commit making it protected. Can you please check this PR again?

Reply:
Thanks, @rodrigoprimo! I originally meant updating the property already defined in `ActionScheduler_Abstract_Schedule` (which this class inherits from), but this also works and is perhaps a safer/more conservative change to make. Looks good :+1: